core_v2/sources/domains.move

```diff
@@ -727,15 +729,24 @@ module aptos_names::domains {
     fun set_reverse_lookup_internal(
         account: &signer,
         token_addr: address,
-    ) acquires NameRecordV2, ReverseLookupRegistryV1, SetReverseLookupEventsV1 {
+    ) acquires CollectionCapabilityV2, NameRecordV2, ReverseRecord, SetReverseLookupEventsV1 {
         let account_addr = signer::address_of(account);
         let record = borrow_global<NameRecordV2>(token_addr);
         let record_obj = object::address_to_object<NameRecordV2>(token_addr);
         assert!(object::owns(record_obj, account_addr), error::permission_denied(ENOT_AUTHORIZED));
 
-        assert!(exists<ReverseLookupRegistryV1>(@aptos_names), error::invalid_state(EREVERSE_LOOKUP_NOT_INITIALIZED));
-        let registry = &mut borrow_global_mut<ReverseLookupRegistryV1>(@aptos_names).registry;
-        table::upsert(registry, account_addr, token_addr);
+        let reverse_record_addr = reverse_record_addr(account_addr);
+        if (!is_object(reverse_record_addr)) {
+            // TODO: Use object ExtendRef
+            let constructor_ref = object::create_named_object(&get_token_signer(), bcs::to_bytes(&account_addr));
```

Comment:
how do we choose the seed here? is this a common practice to just use `bcs::to_bytes(&account_addr)` to reserve an address?